ospfd/ospf_vty.c

```diff
@@ -3365,6 +3365,56 @@ DEFUN (show_ip_ospf_instance,
 	return ret;
 }
 
+static void ospf_interface_auth_show(struct vty *vty, struct ospf_interface *oi,
+				     json_object *json, bool use_json)
+{
+	int auth_type;
+
+	auth_type = OSPF_IF_PARAM(oi, auth_type);
+
+	switch (auth_type) {
+	case OSPF_AUTH_NULL:
+		if (use_json)
+			json_object_string_add(json, "authType",
+					       "NULL Authentication");
+		else
+			vty_out(vty, "  Authentication NULL is enabled\n");
+		break;
+	case OSPF_AUTH_SIMPLE: {
+		if (use_json)
+			json_object_string_add(json, "authType",
+					       "Simple Password");
+		else
+			vty_out(vty,
+				"  Simple password authentication enabled\n");
+		break;
+	}
+	case OSPF_AUTH_CRYPTOGRAPHIC: {
+		struct crypt_key *ckey;
```

Comment:
Is this needed - other places just produce output based on the auth type?

Reply:
In FRR, Cryptographic authentication considered only if at least one key configured otherwise it is considered as NULL authentication while preparing the ospf packet even md5 enabled. (ref api ospf_auth_type). But for other authentication(simple password), irrespective of authentication data authentication applied. 
So by considering this behaviour the authentication information displayed to user  only if at least one key configured in md5 auth.